util/blockdevice/util.go

```diff
@@ -171,3 +195,14 @@ func HasSystemPresentOrError(obj unstructured.Unstructured) (bool, error) {
 	}
 	return true, nil
 }
+
+// GetDeviceTypeOrError return device type HDD or SSD. If not present
+// it returns an error.
+func GetDeviceTypeOrError(obj unstructured.Unstructured) (string, error) {
+	if obj.GetKind() != string(types.KindBlockDevice) {
```

Comment:
if obj is nil, this will panic

Reply:
I am not using pointer. It will not be nil.
if the object inside unstructured.Unstructured{}  is nil then that case is handled here
https://github.com/kubernetes/kubernetes/blob/4bb8df4464f44090ec1a595fbcaa60288e1e0d82/staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/unstructured/unstructured.go#L104

Reply:
I have added a test case for nil object.